bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/MideaACHandlerFactory.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mideaac.internal;
+
+import static org.openhab.binding.mideaac.internal.MideaACBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.mideaac.internal.dto.CloudsDTO;
+import org.openhab.binding.mideaac.internal.handler.MideaACHandler;
+import org.openhab.core.i18n.UnitProvider;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link MideaACHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Jacek Dobrowolski - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.mideaac", service = ThingHandlerFactory.class)
+public class MideaACHandlerFactory extends BaseThingHandlerFactory {
+
+    private UnitProvider unitProvider;
+    private final HttpClient httpClient;
+    private final CloudsDTO clouds;
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    /**
+     * The MideaACHandlerFactory class parameters
+     * 
+     * @param unitProvider OH unitProvider
+     * @param httpClientFactory OH httpClientFactory
+     */
+    @Activate
+    public MideaACHandlerFactory(@Reference UnitProvider unitProvider, @Reference HttpClientFactory httpClientFactory) {
+        this.unitProvider = unitProvider;
+        this.httpClient = httpClientFactory.getCommonHttpClient();
```

Comment:
I would propose to not cache the httpClient, butt the factory. This way everytime the `createHandler` is called a new clientcan be assigned from the factory.
When done this will result in the rare case the httpclient becomes broken, stoppingt/starting the Thing will create a new handler and thus using a new httpclient.

Reply:
Not exactly sure how to do this

Reply:
1. Change `HttpClient httpClient` to  `HttpClientFactory httpClientFactory` at line 44
2. Change this here `this.httpClient = httpClientFactory.getCommonHttpClient();` to `this.httpClientFactory = httpClientFactory;`
3. line 69 change `httpClient` to `httpClientFactory.getCommonHttpClient()`

Reply:
Thanks.  Done